base/libgit2/utils.jl

```diff
@@ -43,3 +43,15 @@ function features()
     end
     return res
 end
+
+"""
+    LibGit2.posixpath(path)
+
+Standardise the path string `path` to use POSIX separators.
+"""
+function posixpath end
+if is_windows()
+    posixpath(path) = replace(path,'\\','/')
```

Comment:
Actually, it seems that is valid, at least for `open`.

The only exception appears to be with ["long" paths](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx):

> Note  File I/O functions in the Windows API convert "/" to "\" as part of converting the name to an NT-style name, except when using the "\\?\" prefix as detailed in the following sections.